core/domain/state_domain.py

```diff
@@ -1461,6 +1466,11 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
                     'Found a duplicate content id %s' % solution_content_id)
             content_id_list.append(solution_content_id)
 
+        if not isinstance(self.solicit_answer_details, bool):
+            raise utils.ValidationError(
```

Comment:
Do we expect `solicit_answer_details` only for specific interactions?

Reply:
Yeah currently it will be true for 'TextInput' interaction only. Also I have added a validation for it in the validate function in the state class.